src/map/utils/petutils.cpp

```diff
@@ -1400,6 +1400,10 @@ namespace petutils
 		PPet->SetMJob(JOB_DRG);
 		PPet->SetMLevel(PMaster->GetMLevel());
 
+        if (PMaster->objtype == TYPE_PC) {

```

Comment:
You actually don't even need this condition. This method is only used by players.